src/Clapi/Valuespace.hs

```diff
@@ -353,6 +370,10 @@ validateChildKeyTypes getType' np orderedKeys expectedTypes =
         (traverse . traverse) getType' expectedTypeMap
     failTypes taggedBadTypes
 
+childTypesOf :: Definition -> Path.Name -> Maybe TypePath
+childTypesOf (TupleDef _ _ _ _) = const Nothing
+childTypesOf (StructDef _ names types _) = flip Map.lookup $ Map.fromList $ zip names types
```

Comment:
Rather than bother to convert to a `Map` and back, you can just do a regular `lookup :: Eq a => a -> [(a, b)] -> Maybe b`